Bmx.Idp.Okta/OktaClient.cs

```diff
@@ -75,10 +79,28 @@ public class OktaClient : IIdentityProvider {
 			return _oktaApps.Where( app => app.AppName == accountType ).Select( app => app.Label ).ToArray();
 		}
 
+		public async Task<string> GetServiceProviderSaml( int selectedAccountIndex ) {
+			var account = _oktaApps[selectedAccountIndex];
+			var accountPage = await GetAccountOkta( new Uri( account.LinkUrl ) );
+			return ExtractAwsSaml( accountPage );
+		}
+
 		private void AddSession( string sessionId ) {
 			_cookieContainer.Add( new Cookie( "sid", sessionId, "/", _httpClient.BaseAddress.Host ) );
 		}
 
+		private string ExtractAwsSaml( string htmlResponse ) {
+			// HTML page is fairly malformed, grab just the <input> with the SAML data for further processing
+			var inputRegexPattern = "<input name=\"SAMLResponse\" type=\"hidden\" value=\".*\"/>";
+			var inputRegex = new Regex( inputRegexPattern, RegexOptions.Compiled );
+
+			// Access the SAML data from the parsed <input>
+			var inputXml = new XmlDocument();
+			inputXml.LoadXml( inputRegex.Match( htmlResponse ).Value );
+
+			return inputXml.SelectSingleNode( "//@value" ).InnerText;
```

Comment:
Thoughts on parsing the SAML response earlier, or adjusting the naming? It's not clear beyond this point that we're passing around SAML response statements until we parse it in `SetSamlToken()`.

Reply:
I should clarify the variable names here, Identity provider at this point is dealing with HTML (inputXML) is an HTML<input> node. 

I used XPath parsing  at this point since it was a straightforward way to get the contents of the <input> node's Value attribute.

As to why Identity Provider is passing the token out as a encoded string, initially I had it setup where it did pass parsed SAML as an XML doc, but it requires writing a lot of additional things to re-encode and export the XMLDocument back to a string that the AWS API accepts.